src/crh/poseidon/mod.rs

```diff
@@ -31,191 +49,210 @@ pub trait PoseidonRoundParams<F: PrimeField>: Default + Clone {
 
 /// The Poseidon permutation.
 #[derive(Default, Clone)]
-pub struct Poseidon<F, P> {
-    pub params: P,
+pub struct PoseidonParameters<F> {
     /// The round key constants
     pub round_keys: Vec<F>,
     /// The MDS matrix to apply in the mix layer.
     pub mds_matrix: Vec<Vec<F>>,
 }
 
-impl<F: PrimeField, P: PoseidonRoundParams<F>> Poseidon<F, P> {
-    fn permute(&self, input: &[F]) -> Vec<F> {
-        let width = P::WIDTH;
-        assert_eq!(input.len(), width);
+impl<F: PrimeField> PoseidonParameters<F> {
+    pub fn new(round_keys: Vec<F>, mds_matrix: Vec<Vec<F>>) -> Self {
+        Self {
+            round_keys,
+            mds_matrix,
+        }
+    }
 
-        let full_rounds_beginning = P::FULL_ROUNDS_BEGINNING;
-        let partial_rounds = P::PARTIAL_ROUNDS;
-        let full_rounds_end = P::FULL_ROUNDS_END;
+    pub fn generate<R: Rng>(rng: &mut R) -> Self {
+        Self {
+            round_keys: Self::create_round_keys(rng),
+            mds_matrix: Self::create_mds(rng),
+        }
+    }
+    pub fn create_mds<R: Rng>(_rng: &mut R) -> Vec<Vec<F>> {
+        let mds_matrix = Vec::new();
+        mds_matrix
```

Comment:
I presume this is just a placeholder for now? If so we should add a `todo!()` here.